### PR TITLE
Format changes, and rename to steps array

### PR DESCRIPTION
**Pull request by jonmv**
@hakonhall please review and merge. Check the `expeced-suite.json` for sample output with lots happening.

### CODE REVIEW

File: client/go/cmd/testdata/tests/expected-suite.out [hunk 1]
```diff
@@ -259,11 +271,52 @@ Expected JSON at /root/children/0/fields/artist ("Boo Fighters") does not match
...
+  }
+}
+
+
+Failed 8 of 9 tests:
+testdata/tests/system-test/wrong-bool-value.json: step 1: Unexpected value at /root/coverage/full: true
+testdata/tests/system-test/wrong-element-count.json: step 1: Unexpected number of elements at /root/children: 1
+testdata/tests/system-test/wrong-field-name.json: step 1: Missing expected field at /root/fields/totalCountDracula
+testdata/tests/system-test/wrong-float-value.json: step 1: Unexpected value at /root/children/0/relevance: 0.38186238359951247
+testdata/tests/system-test/wrong-int-value.json: step 1: Unexpected value at /root/fields/totalCount: 1
```
Comment (hakonhall) [inline client/go/cmd/testdata/tests/expected-suite.out:319]: "Unexpected value" -> "Unexpected response value" may make it slightly clearer? Up to you.

File: client/go/cmd/testdata/tests/expected-suite.out [hunk 2]
```diff
@@ -1,7 +1,9 @@
-Running testdata/tests/system-test/test.json: .... OK!
-Running testdata/tests/system-test/wrong-bool-value.json:
-Failed verifying assertion 0:
-Expected JSON at /root/coverage/full (false) does not match actual (true). Response body:
+Running my test: .... OK
```
Comment (hakonhall) [inline client/go/cmd/testdata/tests/expected-suite.out:1]: Could you add another successful test right after this?  I just want to verify there is no empty line between two successful tests.

**Review by hakonhall — APPROVED** [review]

File: client/go/cmd/testdata/tests/expected-suite.out (hunk 2 above)
Comment (jonmv) [reply client/go/cmd/testdata/tests/expected-suite.out:1]: Hmm, there is an empty line between all tests. Would you rather it weren't?

File: client/go/cmd/testdata/tests/expected-suite.out (hunk 1 above)
Comment (jonmv) [reply client/go/cmd/testdata/tests/expected-suite.out:319]: I think it's better without.

File: client/go/cmd/testdata/tests/expected-suite.out (hunk 2 above)
Comment (hakonhall) [reply client/go/cmd/testdata/tests/expected-suite.out:1]: It's nice with an empty line after an unsuccessful test, because the error message for the test is itself multiline, so it would be difficult to see where the next test begins.

Comment (hakonhall) [reply client/go/cmd/testdata/tests/expected-suite.out:1]: But between successful tests there should be no empty lines? You want

```
Running foo.json: ....ok
Running bar.json: ....ok
Running zoo.json: ....ok
```
